app/vulncode_app/test_api.py

```diff
@@ -0,0 +1,165 @@
+                  "reference": [
+                     {
+                       "reference_id": "CVE-2009-2458",
+                       "source": "",
+                       "url": ""
+                     }
+                     ]}}]
+            },
+            {
+              "name": "mimetex",
+              "version": "1.50-1.1",
+              "vulnerabilities": [{
+                "vulnerability": {
+                  "summary": "Multiple unspecified vulnerabilities in mimeTeX.",
+                  "reference": [
+                     {
+                       "reference_id": "CVE-2009-2459",
+                       "source": "",
+                       "url": ""
+                     }
+                     ]}}]
+            }
+          ]
+
+        self.assertEqual(expected, response.data)
+
+    def test_ubuntu_response(self):
+        with open('tests/test_data/ubuntu_main.html') as f:
+            test_data = f.read()
+
+        extract_data = ubuntu.extract_cves(test_data)
+        ubuntu_dump(extract_data)
+        response = self.client.get('/vulncode_app/data/automake', format='json')
+
+        expected = [
+            {
+              "name": "automake",
+              "version": "",
+              "vulnerabilities": [
+               {
+                 "vulnerability": {
+                   "summary": "",
+                   "reference": [
+                    {
+                        "reference_id": "CVE-2012-3386",
+                        "source": "",
+                        "url": ""
+                    }
+                     ]}}]
+            }
+        ]
+
+        self.assertEqual(expected, response.data)
+
+    def test_blank_response(self):
+        response_invalid = self.client.get('/vulncode_app/data/', format='json')
+        response_blank = self.client.get('/vulncode_app/data/abbpcc', format='json')
+
+        self.assertEqual(404, response_invalid.status_code)
+        self.assertEqual([], response_blank.data)
+
+
+class TestSerializers(TestCase):
+    def test_serializers(self):
+        with open('tests/test_data/debian.json') as f:
+            test_data = json.loads(f.read())
+        extract_data = debian.extract_vulnerabilities(test_data)
+        debian_dump(extract_data)
+
+        pk = Package.objects.filter(name="mimetex")
+        response = PackageSerializer(pk, many=True).data
+
+        expected = [OrderedDict([
+                       ('name', 'mimetex'),
+                       ('version', '1.50-1.1'),
+                       ('vulnerabilities',
+                        [OrderedDict([
+                            ('vulnerability',
+                             OrderedDict([
+                               (
+                                 'summary',
+                                 'Multiple stack-based buffer overflows in mimetex.cgi in mimeTeX'
+                               ),
+                               ('reference',
```

Comment:
Should not this plural "references"?

Reply:
atm, every `reference` contains a single `reference id`. So, I think not.

Reply:
My point is that there are multiple references, hence the list should be plural